### PR TITLE
Nano: Support patching original model'attributes to traced and quantized model

### PR DESCRIPTION
**Pull request by MeouSker77**
## Description

Support accessing original model's attributes after trace and quantization

### 1. Why the change?

Fix issue 3 of [#242](https://github.com/analytics-zoo/nano/issues/242)

### 2. User API changes

N/A

### 3. Summary of the change 

- Patch original model'attributes to traced and quantized model
- Recover the `input_spec` parameter of `InferenceOptimizer.quantize` and `InferenceOptimizer.optimize`

### 4. How to test?
- [ ] N/A
- [ ] Unit test
- [ ] Application test
- [ ] Document test
- [ ] ...

### CODE REVIEW

File: python/nano/src/bigdl/nano/tf/utils.py [hunk 1]
```diff
@@ -13,7 +13,70 @@
...
+    :return: `target_obj`
+    """
+    if inspect.ismethod(target_obj.__setattr__):
+        # `target_obj` has custom `__setattr__`
+        wrapper_type = _create_wrapper_type(type(target_obj), target_obj, source_obj)
+        wrapper_obj = wrapper_type.__new__(wrapper_type)
+        return wrapper_obj
+    else:
+        # `target_obj` has no custom `__setattr__`
+        for name in set(dir(target_obj)) - set(dir(target_obj)):
```
Comment (TheaperDeng) [inline python/nano/src/bigdl/nano/tf/utils.py:70]: should this one be `set(dir(source_obj)) - set(dir(target_obj))` ?

Comment (MeouSker77) [reply python/nano/src/bigdl/nano/tf/utils.py]: yes, fix it

File: python/nano/src/bigdl/nano/tf/utils.py [hunk 2]
```diff
@@ -13,7 +13,70 @@
...
+
+def patch_attrs(target_obj: object, source_obj: object) -> object:
+    """
+    Patch attributes of `source_obj` to `target_obj`.
+
+    :param target_obj: target object
+    :param source_obj: source object
+    :return: `target_obj`
+    """
+    if inspect.ismethod(target_obj.__setattr__):
```
Comment (TheaperDeng) [inline python/nano/src/bigdl/nano/tf/utils.py:63]: maybe this one will cover all the cases, since all obj has `__setattr__` right?

Comment (MeouSker77) [reply python/nano/src/bigdl/nano/tf/utils.py:63]: no, inspect.ismethod('__setattr__') is `True` only when `__setattr__` is defined explicitly by user

Comment (TheaperDeng) [reply python/nano/src/bigdl/nano/tf/utils.py:63]: Yeah, I had a try you are right

**Review by TheaperDeng — APPROVED** [review]
LGTM

**Pull request merged**